bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/discovery/InsteonLegacyDiscoveryService.java

```diff
@@ -59,13 +59,16 @@ public void addInsteonDevices(List<InsteonAddress> addresses) {
             ThingUID bridgeUID = handler.getThing().getUID();
             String id = address.toString().replace(".", "");
             ThingUID thingUID = new ThingUID(THING_TYPE_LEGACY_DEVICE, bridgeUID, id);
+            ThingUID oldThingUID = new ThingUID(THING_TYPE_DEVICE, bridgeUID, id);
```

Comment:
This seems the wrong way around. I would assume the legacy devices are considered as old and should be removed?

Reply:
Old in terms of Thing UID. Newly discovered legacy device should use the new Thing UID format. If it makes sense.